service/history/workflow/cache/cache.go

```diff
@@ -95,23 +100,26 @@ const (
 	workflowLockTimeoutTailTime = 500 * time.Millisecond
 )
 
-func NewCache(shard shard.Context) Cache {
+func NewCache(
+	config *configs.Config,
+	logger log.Logger,
+	metricsHandler metrics.Handler,
+) Cache {
 	opts := &cache.Options{}
-	config := shard.GetConfig()
 	opts.TTL = config.HistoryCacheTTL()
 	opts.Pin = true
 
 	return &CacheImpl{
 		Cache:                     cache.New(config.HistoryCacheMaxSize(), opts),
-		shard:                     shard,
-		logger:                    log.With(shard.GetLogger(), tag.ComponentHistoryCache),
-		metricsHandler:            shard.GetMetricsHandler().WithTags(metrics.CacheTypeTag(metrics.MutableStateCacheTypeTagValue)),
+		logger:                    log.With(logger, tag.ComponentHistoryCache),
```

Comment:
plz add shardID tag (and hostIdentity I think?) whenever this logger is used.

Reply:
Updated in workflow context